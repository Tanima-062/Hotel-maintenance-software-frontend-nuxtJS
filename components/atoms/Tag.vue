<template>
  <div
    class="border border-solid rounded-md select-none"
    :class="`${outlineColor} ${py} ${textSize} ${fontWeight} ${padding} ${clickablePointer}`"
    @click="click"
    @mousedown="mousedown"
  >
    <slot />
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: 'white' })
  color!: string

  @Prop({ default: 'py-0.5' })
  py!: string

  @Prop({ default: 'text-sm' })
  textSize!: string

  @Prop({ default: 'font-hairline' })
  fontWeight!: string

  @Prop({ default: 'px-2' })
  padding!: string

  @Prop({ default: false })
  clickable!: boolean

  get clickablePointer(): string {
    return this.clickable ? 'cursor-pointer' : ''
  }

  get outlineColor(): string {
    switch (true) {
      case this.color === 'blue':
        return 'border-tagColor-blue text-tagColor-blue'
      case this.color === 'green':
        return 'border-tagColor-green text-tagColor-green'
      case this.color === 'yellow':
        return 'border-tagColor-yellow text-tagColor-yellow'
      case this.color === 'red':
        return 'border-tagColor-red text-tagColor-red'
      case this.color === 'cancel':
        return 'border-tagColor-cancel text-tagColor-cancel'
      case this.color === 'purple':
        return 'border-tagColor-purple text-tagColor-purple'
      default:
        return 'border-tagColor text-tagColor'
    }
  }

  @Emit()
  private click(e: Event): Event {
    return e
  }

  @Emit()
  private mousedown(e: Event): Event {
    return e
  }
}
</script>
