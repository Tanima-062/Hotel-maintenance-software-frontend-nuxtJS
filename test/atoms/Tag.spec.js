import { shallowMount } from '@vue/test-utils'
import Tag from '@/components/atoms/Tag.vue'

describe('Atoms Tag Check', () => {
  test('Tag Test', () => {
    const wrapper = shallowMount(Tag)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
