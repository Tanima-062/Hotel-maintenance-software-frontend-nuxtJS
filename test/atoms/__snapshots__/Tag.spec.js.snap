// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Tag Check Tag Test 1`] = `
<div
  class="border border-solid rounded-md select-none border-tagColor text-tagColor py-0.5 text-sm font-hairline px-2 "
/>
`;
